### PR TITLE
[API-39196] Add `transaction_id` to 526 synchronous logs

### PR DESCRIPTION
**Pull request by tycol7**
## Summary

Add `transaction_id` to:
* Docker container service logs
* PDF generation service logs
* EVSS service logs

## Related issue(s)

[API-39196](https://jira.devops.va.gov/browse/API-39196)

## Testing done

- [x] *New code is covered by unit tests*

## Screenshots

N/A

## What areas of the site does it impact?

526 synchronous endpoint, including:
* Docker container service
* PDF generation service
* EVSS service

## Acceptance criteria

- [x]  I added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

N/A

### CODE REVIEW

File: modules/claims_api/app/services/claims_api/disability_compensation/docker_container_service.rb [hunk 1]
```diff
@@ -10,21 +10,21 @@ class DockerContainerService < ServiceBase
       LOG_TAG = '526_v2_Docker_Container_service'
 
       def upload(claim_id)
-        log_job_progress(claim_id,
-                         'Docker container service started')
-
         auto_claim = get_claim(claim_id)
+
+        log_job_progress(claim_id, 'Docker container service started', auto_claim.transaction_id)
```
Comment (stiehlrod) [inline modules/claims_api/app/services/claims_api/disability_compensation/docker_container_service.rb:15]: @tycol7 I am not seeing the transaction_id
Screenshot 2024-08-29 at 8 40 29 AM

File: modules/claims_api/app/services/claims_api/disability_compensation/docker_container_service.rb [hunk 2]
```diff
@@ -10,21 +10,21 @@ class DockerContainerService < ServiceBase
...
+
         update_auth_headers(auto_claim) if auto_claim.transaction_id.present?
 
         evss_data = get_evss_data(auto_claim)
 
-        log_job_progress(claim_id,
-                         'Submitting mapped data to Docker container')
+        log_job_progress(claim_id, 'Submitting mapped data to Docker container', auto_claim.transaction_id)
```
Comment (stiehlrod) [inline modules/claims_api/app/services/claims_api/disability_compensation/docker_container_service.rb:21]: Nor here
Screenshot 2024-08-29 at 8 43 10 AM

File: modules/claims_api/app/services/claims_api/disability_compensation/pdf_generation_service.rb [hunk 3]
```diff
@@ -12,20 +12,19 @@ class PdfGenerationService < ServiceBase
...
+        log_job_progress(auto_claim.id, "526EZ PDF generator started for claim #{auto_claim.id}",
+                         auto_claim.transaction_id)
 
         mapped_claim = generate_mapped_claim(auto_claim, middle_initial)
         pdf_string = generate_526_pdf(mapped_claim)
 
         if pdf_string.empty?
-          log_job_progress(auto_claim.id,
-                           '526EZ PDF generator failed to return PDF string for claim')
+          log_job_progress(auto_claim.id, '526EZ PDF generator failed to return PDF string for claim',
```
Comment (stiehlrod) [inline modules/claims_api/app/services/claims_api/disability_compensation/pdf_generation_service.rb:22]: Screenshot 2024-08-29 at 8 44 27 AM

File: modules/claims_api/app/services/claims_api/disability_compensation/pdf_generation_service.rb [hunk 4]
```diff
@@ -35,17 +34,16 @@ def generate(claim_id, middle_initial) # rubocop:disable Metrics/MethodLength
                                                             tempfile: File.open(path)
                                                           })
```
Comment (stiehlrod) [inline modules/claims_api/app/services/claims_api/disability_compensation/pdf_generation_service.rb:36]: Screenshot 2024-08-29 at 8 44 27 AM

File: modules/claims_api/app/services/claims_api/disability_compensation/pdf_generation_service.rb [hunk 5]
```diff
@@ -35,17 +34,16 @@ def generate(claim_id, middle_initial) # rubocop:disable Metrics/MethodLength
...
 
           auto_claim.set_file_data!(upload, EVSS_DOCUMENT_TYPE)
           auto_claim.save!
 
           ::Common::FileHelpers.delete_file_if_exists(path)
         end
 
-        log_job_progress(auto_claim.id,
-                         '526EZ PDF generator job finished')
+        log_job_progress(auto_claim.id, '526EZ PDF generator job finished', auto_claim.transaction_id)
```
Comment (stiehlrod) [inline modules/claims_api/app/services/claims_api/disability_compensation/pdf_generation_service.rb:46]: Screenshot 2024-08-29 at 8 45 55 AM

**Comment by stiehlrod** [comment]
Hey @tycol7, I am not seeing the transaction_id come through in the logs.
Screenshot 2024-08-29 at 8 50 35 AM

**Comment by tycol7** [comment]
> Hey @tycol7, I am not seeing the transaction_id come through in the logs. Screenshot 2024-08-29 at 8 50 35 AM

Did your `claim` have a `transaction_id`? We will omit `transaction_id` if not on the claim.

**Comment by stiehlrod** [comment]
> > Hey @tycol7, I am not seeing the transaction_id come through in the logs. Screenshot 2024-08-29 at 8 50 35 AM
> 
> Did your `claim` have a `transaction_id`? We will omit `transaction_id` if not on the claim.

Got it, my apologies.

**Review by stiehlrod — APPROVED** [review]

**Review by rjohnson2011 — APPROVED** [review]

**Pull request merged**